### PR TITLE
[Bug 1399639] Updating Django Rest Framework

### PR DESCRIPTION
**Pull request by safwanrahman**
It passes all the tests! But need some more investigation.
I will investigate more later this week.
@jwhitlock Can you take a look if I have taken proper approach?

### CODE REVIEW

File: requirements/default.txt [hunk 1]
```diff
@@ -113,9 +113,9 @@ django-recaptcha==1.0.5 \
     --hash=sha256:d4d896dd399a1a0810af8e59a646bb5a1586b2cc8b8f32d16766ade5fba79475
 
 # RESTful API framework
-djangorestframework==3.3.2 \
-    --hash=sha256:5634b1ff56581bf0fe4075e86227fc9693c1ca031c7213c9ae942c445c24817b \
-    --hash=sha256:4962418a57804f8323282728a4f9b9496e78caec1adda352170697752eff01bf
+djangorestframework==3.5.4 \
+    --hash=sha256:110afa12784ceadfb50808882689302d266785b51e3d13286744333ff6d78e60 \
+    --hash=sha256:f995a35ae22f354d2a9a42ee6d2c059c101f826b1485ed46781677895ad25ee5
```
Comment (jwhitlock) [inline requirements/default.txt]: I'm adding more links to the requirements files, see https://github.com/mozilla/kuma/blob/master/requirements/README.rst#requirements-format.  Here's what I have:

```
# RESTful API framework, used by search
# Code: https://github.com/encode/django-rest-framework
# Changes: http://www.django-rest-framework.org/topics/release-notes/
# Docs: http://www.django-rest-framework.org
djangorestframework==3.5.4 \
    --hash=sha256:110afa12784ceadfb50808882689302d266785b51e3d13286744333ff6d78e60 \
    --hash=sha256:f995a35ae22f354d2a9a42ee6d2c059c101f826b1485ed46781677895ad25ee5
```

**Comment by jwhitlock** [comment]
Thanks for tackling this @safwanrahman! I appreciate that you are limiting changes to those needed for the upgrade. This is quick feedback, I'll get into a detailed review later today.

Tests passing is a good sign, although I really don't trust the tests around search 😬 . I'll read the release notes and poke around to see if these look good.

I'm taking an incremental approach with these updates, shipping one or two packages at a time. ``django-tidings`` is the next one (PR #4660), going out today, so I'll wait to merge until that is deployed. [bug 1399639](https://bugzilla.mozilla.org/show_bug.cgi?id=1399639) is the bug to use for package upgrades.

I think DRF 3.6 supports Django 1.8 still, and if so, we should get to that eventually. I'm happy to focus on DRF 3.5, then do DRF 3.6 in the next PR, if there are a lot of changes for DRF 3.6.

File: requirements/default.txt (hunk 1 above)
Comment (jwhitlock) [reply requirements/default.txt]: I've tested DRF 3.6.4, and that works with these changes as well. Please switch to that version.

File: kuma/search/renderers.py [hunk 2]
```diff
@@ -9,40 +9,14 @@ class ExtendedTemplateHTMLRenderer(TemplateHTMLRenderer):
...
-        if response.exception:
-            template = self.get_exception_template(response)
-        else:
-            template_names = self.get_template_names(response, view)
-            template = self.resolve_template(template_names)
+        data.update(new_data)
 
-        context = self.resolve_context(data, request, response)
-        return template.render(context, request=request)
+        return super(ExtendedTemplateHTMLRenderer, self).get_template_context(data, renderer_context)
```
Comment (jwhitlock) [inline kuma/search/renderers.py]: 😄 good job resolving this TODO, the code is much cleaner.

**Comment by safwanrahman** [comment]
@jwhitlock Updated with the comments.

**Pull request merged**